src/Workspaces/Core/Portable/Classification/EmbeddedLanguages/ExportEmbeddedLanguageClassifier.cs

```diff
@@ -12,16 +14,55 @@ namespace Microsoft.CodeAnalysis.Classification
     /// </summary>
     [MetadataAttribute]
     [AttributeUsage(AttributeTargets.Class)]
-    internal sealed class ExportEmbeddedLanguageClassifier : ExportAttribute
+    internal class ExportEmbeddedLanguageClassifierAttribute : ExportAttribute
     {
+        /// <summary>
+        /// Name of the classifier.
+        /// </summary>
         public string Name { get; }
+
+        /// <summary>
+        /// Name of the containing language hosting the embedded language.  e.g. C# or VB.
+        /// </summary>
         public string Language { get; }
 
-        public ExportEmbeddedLanguageClassifier(string name, string language)
+        /// <summary>
+        /// Identifiers in code (or StringSyntaxAttribute) used to identify an embedded language string. For example
+        /// <c>Regex</c> or <c>Json</c>.
+        /// </summary>
+        /// <remarks>This can be used to find usages of an embedded language using a comment marker like <c>//
+        /// lang=regex</c> or passed to a symbol annotated with <c>[StringSyntaxAttribyte("Regex")]</c>.  The identifier
+        /// is case sensitive for the StringSyntaxAttribute, and case insensitive for the comment.
+        /// </remarks>
+        public ImmutableArray<string> Identifiers { get; }
+
+        public ExportEmbeddedLanguageClassifierAttribute(
+            string name, string language, params string[] identifiers)
             : base(typeof(IEmbeddedLanguageClassifier))
         {
             Name = name ?? throw new ArgumentNullException(nameof(name));
             Language = language ?? throw new ArgumentNullException(nameof(language));
+            Identifiers = identifiers.ToImmutableArray();
+        }
+    }
+
+    /// <summary>
+    /// Internal version of ExportEmbeddedLanguageClassifierAttribute.  Used so we can allow regex/json to still light
+    /// up on legacy APIs not using the new [StringSyntax] attribute the runtime added.  For public extensions that's
+    /// the only mechanism we support.
+    /// </summary>
+    [MetadataAttribute]
+    [AttributeUsage(AttributeTargets.Class)]
+    internal sealed class ExportEmbeddedLanguageClassifierInternalAttribute : ExportEmbeddedLanguageClassifierAttribute
```

Comment:
This will NOT become public.  It's a backdoor for us to classify legacy APIs.

Reply:
Is the distinction worth it? Seems simpler just to allow both forms of specifying a language.